packages/starlight/__e2e__/test-utils.ts

```diff
@@ -9,27 +9,63 @@ process.env.ASTRO_DISABLE_UPDATE_CHECK = 'true';
 
 // Setup a test environment that will build and start a preview server for a given fixture path and
 // provide a Starlight Playwright fixture accessible from within all tests.
-export async function testFactory(fixturePath: string) {
-	let previewServer: PreviewServer | undefined;
+export function testFactory(fixturePath: string) {
+	const fixturePathUrl = new URL(fixturePath, import.meta.url);
+	// Combining absolute paths with a `file:` base URL on Windows results
+	// in a URL with the drive letter as the protocol.
+	// In that case, use the URL as is instead of interpreting the `file:` protocol.
+	const root =
+		fixturePathUrl.protocol === 'file:'
+			? fileURLToPath(new URL(fixturePath, import.meta.url))
+			: fixturePathUrl.toString();
 
-	const test = baseTest.extend<{ starlight: StarlightPage }>({
-		starlight: async ({ page }, use) => {
-			if (!previewServer) {
-				throw new Error('Could not find a preview server to run tests against.');
-			}
+	async function makeServer(
+		options: {
+			mode?: 'build' | 'dev';
+		} = {}
+	): Promise<Server> {
+		const { mode } = options;
+		if (mode === 'dev') {
+			return await dev({
+				logLevel: 'error',
+				root,
+				vite: { optimizeDeps: { noDiscovery: true } },
```

Comment:
Let's add a small comment that this was necessary to avoid some timeouts.

Now that it's confirmed that this was the issue, when I get the chance to make a small repro and open an issue on the Vite repo, I'll also make a new PR to add the link here too.